packages/expo-updates/src/UpdatesHooks.ts

```diff
@@ -44,3 +44,35 @@ export const useUpdateEvents = (listener: (event: UpdateEvent) => void) => {
     return undefined;
   }, []);
 };
+
+/**
+ * @hidden
+ */
+export const useUpdatesState: () => UpdatesNativeStateMachineContext = () => {
+  // Hook to return the Updates state machine context maintained
+  // in native code.
+  // Eventually, this will be used to construct the information returned by `useUpdates()`.
+  // Used internally by this module and not exported publicly.
+  const [localState, setLocalState] = useState<UpdatesNativeStateMachineContext>({
+    isUpdateAvailable: false,
+    isUpdatePending: false,
+    isRollback: false,
+    isChecking: false,
+    isDownloading: false,
+    isRestarting: false,
+    checkError: undefined,
+    downloadError: undefined,
+    latestManifest: undefined,
+    downloadedManifest: undefined,
+  });
+  useEffect(() => {
+    Updates.nativeStateMachineContext().then((context) => {
+      setLocalState(context);
+    });
+    const subscription = Updates.addUpdatesStateChangeListener((event) => {
+      setLocalState(() => event.context);
```

Comment:
```suggestion
      setLocalState(event.context);
```

Reply:
same comment as last time. the two are equivalent. One only needs to provide a function here if merging is desired and then the parameter for the previous would be used.